editor/src/components/inspector/common/longhand-shorthand-hooks.ts

```diff
@@ -245,14 +231,92 @@ export function useInspectorInfoLonghandShorthand<
       controlStyles: getControlStyles(controlStatus),
       onSubmitValue: onSubmitValue,
       onTransientSubmitValue: onTransientSubmitValue,
-      onUnsetValues: onUnsetValues,
+      onUnsetValues: NO_OP,
       orderedPropKeys: orderedPropKeys,
     }
   })
 
-  return mapArrayToDictionary(
+  const longhandResultsDictionary = mapArrayToDictionary(
     longhandResults,
     (_, index) => longhands[index],
     (result) => result,
   )
+
+  const longhandResultsWithUnset = objectMap((longhandResult, longhandToUnset) => {
+    const onUnsetValues = () => {
+      const allPropKeysEqual = allOrderedPropKeys.every((propKeys) => {
+        return arrayEquals(propKeys, allOrderedPropKeys[0])
+      })
+      if (!allPropKeysEqual) {
+        // we do nothing for now. we cannot ensure that we can make a sensible update and surface it on the UI as well
+        return
+      }
+      const propkeysToUse = allOrderedPropKeys[0]
+
+      const actionsToDispatch = (() => {
+        const shorthandExists = propkeysToUse.includes(shorthand)
+        const longhandPropertyPath = pathMappingFn(longhandToUnset, inspectorTargetPath)
+        const shorthandPropertyPath = pathMappingFn(shorthand, inspectorTargetPath)
+
+        if (!shorthandExists) {
+          return flatMapArray((selectedView) => {
+            if (isInstancePath(selectedView)) {
+              return [unsetProperty(selectedView, longhandPropertyPath)]
+            } else {
+              return []
+            }
+          }, selectedViewsRef.current)
+        } else {
+          const shorthandControlled = shorthandInfo.propertyStatus.controlled
+          if (shorthandControlled) {
+            // we want to remove existing uses of the longhand, and append a `{[longhand]: undefined}` at the end of the object
+            return flatMapArray((selectedView) => {
+              if (isInstancePath(selectedView)) {
+                return [
+                  unsetProperty(selectedView, longhandPropertyPath),
+                  setProp_UNSAFE(
+                    selectedView,
+                    longhandPropertyPath,
+                    jsxAttributeValue(undefined, emptyComments),
+                  ),
+                ]
+              } else {
+                return []
+              }
+            }, selectedViewsRef.current)
+          } else {
+            // we have to split the shorthand into longhands 😭
+            return flatMapArray((selectedView) => {
+              if (isInstancePath(selectedView)) {
+                return [
+                  unsetProperty(selectedView, longhandPropertyPath),
+                  unsetProperty(selectedView, shorthandPropertyPath),
+                  ...longhands
```

Comment:
Could this be broken out into a variable first for clarity?

Reply:
fixed